docs/manual/idl.rst

```diff
@@ -1,36 +1,34 @@
 
 Working with the IDL compiler
 -----------------------------
 
-You use the IDL compiler if you already have an idl file to define your types or if you require interoperability with non-Python projects. The ``idlpy`` library is built as part of the python package leveraging the scikit-build cmake integration. We will soon integrate a locator mechanism into ``idlc`` to retrieve the library location, since the ``idlpy`` library will be part of the python package installation and not on the `LD_LIBRARY_PATH` or equivalent. For now you can employ the same tactic as ``idlc`` will do eventually and use a hidden method:
+Use the IDL compiler if you already have an IDL file that defines your types, or if you require interoperability with non-Python projects. The ``idlpy`` library is built as part of the python package leveraging the ``scikit-build cmake`` integration. 
 
+After installing the CycloneDDS Python backend you can run ``idlc`` with the ``-l py`` flag to generate Python code:
 
 .. code-block:: shell
 
    idlc -l py your_file.idl
 
-If you wish to nest the resulting Python module inside an existing package you can specify the path from the intended root. So if you have a package 'wubble' with a submodule 'fruzzy' and want the generated modules and types under there you can pass ``py-root-prefix``:
+To nest the resulting Python module inside an existing package, you can specify the path from the intended root. If you have a package 'wubble' with a submodule 'fruzzy' and want the generated modules and types under there you can pass ``py-root-prefix``:
 
 .. code-block:: shell
 
    idlc -l py -p py-root-prefix=wubble.fruzzy your_file.idl
 
-
-.. _datatypes:
-
 IDL Datatypes in Python
 -----------------------
 
-The ``cyclonedds.idl`` package implements defining IDL unions and structs and their OMG XCDR-V1 encoding in pure python. There should almost never be a need to delve into the details of this package when using DDS. In most cases the IDL compiler will write the code that uses this package. However, it is possible to write the objects manually. This is of course only useful if you do not plan to interact with other language clients, but makes perfect sense in a python-only project. If you are manually writing IDL objects your most important tool is :class:`IdlStruct<cyclonedds.idl.IdlStruct>`.
+The ``cyclonedds.idl`` package implements the IDL unions, structs, and their OMG XCDR-V1 encoding, in python. In most cases, the IDL compiler writes the code that references this package without the need to edit the objects. However, for python-only projects it is possible to write the objects manually (where cross-language interactions are not required). To manually write IDL objects, use the :class:`IdlStruct<cyclonedds.idl.IdlStruct>` tool.
```

Comment:
"use the :class:`IdlStruct<cyclonedds.idl.IdlStruct>` tool." This class is not a tool, my previous wording was a little awkward. I would say: "To manually write IDL objects you can make your classes inherit from :class:`IdlStruct<cyclonedds.idl.IdlStruct>`, :class:`IdlStruct<cyclonedds.idl.IdlUnion>`, :class:`IdlStruct<cyclonedds.idl.IdlEnum>` or :class:`IdlStruct<cyclonedds.idl.IdlBitmaks>`.

Reply:
Many thanks for the edits